src/components/widgets/Accordion.js

```diff
@@ -0,0 +1,92 @@
+import { useEffect, useRef, useState } from 'react';
+import hljs from 'highlight.js/lib/core';
+import 'highlight.js/styles/night-owl.css';
+import html from 'highlight.js/lib/languages/xml';
+
+function AccordionItem({ title, children }) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleAccordionTitleClick = () => {
+    setIsOpen(!isOpen);
+  };
+
+  return (
+    <div className="accordion-item">
+      <h3 className={`accordion-title ${isOpen ? 'arrow-up' : 'arrow-down'}`} onClick={handleAccordionTitleClick}>{title}</h3>
+      {isOpen && (
+        <div className="accordion-content">
+          {children}
+        </div>
+      )}
+    </div>
+  );
+}
+
+export default function Accordion() {
+  const codeRef = useRef(null);
+  const [buttonText, setButtonText] = useState('Copy code');
+
+  hljs.registerLanguage('html', html);
+  useEffect(() => {
+    hljs.highlightAll();
+  }, []);
+
+  const handleCopyCode = () => {
+    const codeElement = codeRef.current;
+    const range = document.createRange();
+    range.selectNode(codeElement);
+    window.getSelection().removeAllRanges();
+    window.getSelection().addRange(range);
+    document.execCommand('copy');
+    window.getSelection().removeAllRanges();
+
+    setButtonText('Copied!');
+
+    setTimeout(() => {
+      setButtonText('Copy code');
+    }, 2000);
+  };
+
+  return (
+    <section className="wd-content" id="toc-accordion">
+      <h2 id="accordion" className="section-top">Accordion</h2>
+      <p>Use the <strong>Accordion</strong> to display content in collapsible compartments.</p>
+      <p>Use <span className="wd-monospace">&lt;div className="accordion"&gt;&lt;/div&gt;</span> to wrap your content.</p>
+      <div className="wd-window">
+        <div className="wd-visual-ex">
+          <div className="accordion">
+          <AccordionItem title="Read">
+            <p>Making News, March/April 2018. By Deb Venasse, Reporter, <em>IBPA Independent</em> magazine</p>
+          </AccordionItem>
+          <AccordionItem title="Discuss">
+            <p>Adweek is a well-known organization that covers media and communication people, places, and things. Some of its popular elements with journalists, creatives, and media workers are the newsletters with news about the actual people doing the media work.</p>
+          </AccordionItem>
+          </div>
+        </div>
+        <div className="wd-html-code">
+          <pre>
+            <code className="language-html" ref={codeRef}>
+            {String.raw`<div className="accordion">
+  <div className="accordion-item">
```

Comment:
I looked at the accordion widget for [SPA 201](https://d2l.pima.edu/d2l/le/content/482373/viewContent/4438484/View) and [SPA 202](https://d2l.pima.edu/d2l/le/content/482376/viewContent/4826262/View) and it doesn't have this wrapping div.

I think it is a good idea. I tested it in the testing area in themepack and it still seems to work fine.

Reply:
I added the wrapping div because it was needed for course builder so just thought I'd add it here too.